compiler/src/main/java/com/squareup/anvil/compiler/Utils.kt

```diff
@@ -84,3 +83,30 @@ internal fun FqName.isAnvilModule(): Boolean {
 internal fun <T : ClassReference> ClassId.classReferenceOrNull(
   module: ModuleDescriptor
 ): T? = asSingleFqName().toClassReferenceOrNull(module) as T?
+
+/**
+ * Returns the single element matching the given [predicate], or `null` if element was not found.
+ * Unlike [singleOrNull] this method throws an exception if more than one element is found.
+ */
+internal inline fun <T> Iterable<T>.singleOrEmpty(predicate: (T) -> Boolean): T? {
```

Comment:
In the old PR @JoelWilcox suggested renaming this method to `singleOrNull`, but `singleOrNull` already exists in the standard library. I kept `singleOrEmpty` on purpose, because the iterable object must of a single or no element (must be empty).